consensus/types/src/beacon_state.rs

```diff
@@ -2493,6 +2495,22 @@ impl<E: EthSpec> BeaconState<E> {
                     .checked_sub(self.num_fields_pow2())
                     .ok_or(Error::IndexNotSupported(generalized_index))?
             }
+            _ => return Err(Error::IndexNotSupported(generalized_index)),
+        };
+
+        // 2. get beacon state leaves
+        let leaves = self.get_beacon_state_leaves();
+
+        // 3. generate proof
+        self.generate_proof(field_index, &leaves)
+    }
+
+    pub fn compute_finalized_root_proof(
+        &self,
+        generalized_index: usize,
+    ) -> Result<Vec<Hash256>, Error> {
+        // 1. Convert generalized index to field index.
+        let field_index = match generalized_index {
```

Comment:
Just match on the self.fork_unchecked() and pick the right gindex. No need to pass the generalized_index as argument. I guess now that the function is repeated we don't need to list the 1,2,3,4 steps on each function. Same for compute_sync_committee_proof if you split the function into `compute_next_sync_committee_proof` and `compute_current_sync_committee_proof` they are short functions that are no longer fallible. It simplifies the logic on the caller side too